lib/ansible/galaxy/role.py

```diff
@@ -400,4 +400,12 @@ def requirements(self):
 
                     break
 
+            if type(self._requirements) is dict:
```

Comment:
Type checks should use `isinstance` and compare against `Mapping` and not `dict`